### PR TITLE
Test to make sure cached CSFNetwork is dropped on logout

### PR DESCRIPTION
**Pull request by wmathurin**
Also fix for unrelated test

### CODE REVIEW

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkTests/MockNetworkTests.m [hunk 1]
```diff
@@ -57,6 +57,10 @@ - (void)testNetworkBasics {
     CSFSalesforceAction *action1 = [[CSFSalesforceAction alloc] initWithResponseBlock:nil];
     CSFSalesforceAction *action2 = [[CSFSalesforceAction alloc] initWithResponseBlock:nil];
     
+    // Setting verbs and methods (values do not matter - but can't be nil otherwise it confuses duplicateActionInFlight/isEqualToAction
+    action1.method = action2.method = @"POST";
```
Comment (wmathurin) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkTests/MockNetworkTests.m]: That's the fix for the other test - basically isEqualToAction (called by duplicateActionInFlight) doesn't like nil verbs/methods. NB Real life actions would always have a verb and method.

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkTests/MockNetworkTests.m [hunk 2]
```diff
@@ -126,4 +130,13 @@ - (void)dont_testNetworkContexts {
     XCTAssertFalse(action5.cancelled);
 }
 
+// Test that the cached CSFNetwork is removed on logout
+- (void) testLogout {
+    SFUserAccount *user = [TestDataAction testUserAccount];
```
Comment (wmathurin) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkTests/MockNetworkTests.m]: That's the new test, get a CSFNetwork, check cache of CSFNetwork's, logout, check cache of CSFNetwork's again.

**Comment by bhariharan** [comment]
:+1:

**Comment by khawkins** [comment]
:+1:

**Pull request merged**